pkg/storage/context.go

```diff
@@ -0,0 +1,67 @@
+package storage
+
+import (
+	"context"
+
+	openfgapb "go.buf.build/openfga/go/openfga/api/openfga/v1"
+	"go.opentelemetry.io/otel/trace"
+)
+
+// ContextTracerWrapper is a wrapper around a datastore that passes a new
+// context to the underlying datastore methods. It must be the first wrapper
+// around the datastore if traces are to work properly.
+type ContextTracerWrapper struct {
+	OpenFGADatastore
```

Comment:
If we're going to wrap the whole datastore here, should we wrap every method of the datastore? If not, can we just wrap the `RelationshipTupleReader` interface instead?

My fear is that we'll blindly think that the whole datastore is wrapped with the alternative context, but to our surprise it won't be.